tools/trace_analyzer_tool_imp.h

```diff
@@ -0,0 +1,306 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#ifndef ROCKSDB_LITE
+
+#include "rocksdb/trace_analyzer_tool.h"
+
+#include <stdio.h>
+#include <fstream>
+#include <list>
+#include <map>
+#include <queue>
+#include <set>
+#include <utility>
+#include <vector>
+
+#include "rocksdb/env.h"
+#include "rocksdb/write_batch.h"
+#include "util/trace_replay.h"
+
+namespace rocksdb {
+
+class DBImpl;
+class WriteBatch;
+class AnalyzerOptions;
+class TraceWriteHandler;
+
+const int taTypeNum = 7;
+
+enum TraceOperationType : uint32_t {
+  taGet = 0,
+  taPut = 1,
+  taDelete = 2,
+  taSingleDelete = 3,
+  taRangeDelete = 4,
+  taMerge = 5,
+  taIter = 6
+};
+
+struct TraceUnit {
+  uint32_t type;
+  std::string key;
+  size_t value_size;
+  uint64_t ts;
+  uint32_t cf_id;
+};
+
+struct TypeCorre {
+  uint64_t count;
+  uint64_t total_ts;
+};
+
+struct StatsUnit {
+  uint64_t key_id;
+  uint32_t cf_id;
+  size_t value_size;
+  uint64_t access_count;
+  uint64_t latest_ts;
+  uint32_t latest_type;
+  std::vector<TypeCorre> v_corre;
+};
+
+
+class AnalyzerOptions {
+ public:
+  bool output_key_stats;
+  bool output_access_count_stats;
+  bool output_trace_unit;
+  bool output_time_serial;
+  bool output_prefix_cut;
+  bool output_trace_sequence;
+  bool output_io_stats;
+  bool output_correlation;
+  bool input_key_space;
+  bool use_get;
+  bool use_put;
+  bool use_delete;
+  bool use_single_delete;
+  bool use_range_delete;
+  bool use_merge;
+  bool use_iterator;
+  bool no_key;
+  bool print_overall_stats;
+  bool print_key_distribution;
+  bool print_value_distribution;
+  bool print_top_k_access;
+  uint64_t  output_ignore_count;
+  uint64_t start_time;
+  int  value_interval;
+  int top_k;
+  int prefix_cut;
+  std::string output_prefix;
+  std::string key_space_dir;
+  std::vector<std::vector<int>> corre_map;
+  std::vector<std::pair<int, int>> corre_list;
+
+  AnalyzerOptions();
+
+  ~AnalyzerOptions();
+
+  void SparseCorreInput(const std::string& in_str);
+};
+
+
+struct TraceStats {
+  uint32_t cf_id;
+  std::string cf_name;
+  uint64_t a_count;
```

Comment:
What does "a_" mean? Analysis?

Reply:
"a" prefix means "accessed", "w" prefix means "whole". Using the full name makes some variable too long. Should I change them?